Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp

```diff
@@ -13828,7 +13828,7 @@ void SpeculativeJIT::compileRegExpTest(Node* node)
 
 void SpeculativeJIT::compileStringReplace(Node* node)
 {
-    ASSERT(node->op() == StringReplace || node->op() == StringReplaceRegExp);
+    ASSERT(node->op() == StringReplace || node->op() == StringReplaceRegExp || node->op() == StringReplaceAll);
     bool sample = false;
     if (sample)
         incrementSuperSamplerCount();
```

Comment:
Doesn't it allow `StringReplaceAll` to use `operationStringProtoFuncReplaceRegExpString` function?

Reply:
`StringReplaceAll` already uses `operationStringProtoFuncReplaceRegExpString`.

For example, in the benchmark at [this link](https://github.com/WebKit/WebKit/blob/584299b75a8d9305123a5cf3e2c79983e6125747/JSTests/microbenchmarks/string-replaceall-regexp-string-benchmark.js), when `compileStringReplace` is called, `node->op()` is `StringReplaceAll`, and at line 13870, `callOperation(operationStringProtoFuncReplaceRegExpString, ...)` is invoked.